src/Simulation/Common/QubitManager.cs

```diff
@@ -300,8 +306,31 @@ protected virtual void ReleaseOneQubit(Qubit qubit, bool usedOnlyForBorrowing)
                 {
                     throw new ArgumentException("Attempt to free qubit that has not been allocated.");
                 }
-                qubits[qubit.Id] = free;
-                free = qubit.Id;
+                if (EncourageReuse) { 
+                    qubits[qubit.Id] = free;
+                    free = qubit.Id;
+                } 
+                else
+                {
+                    if (MayExtendCapacity)
+                    {
+
```

Comment:
Add a comment _why_ in this case there is nothing to do?

Reply:
Ok, makes sense.